benchmark/rocblas/blas2/trsv.cpp

```diff
@@ -0,0 +1,205 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename trsv.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string uplo, std::string t, std::string diag, int n) {
+  std::ostringstream str{};
+  str << "BM_Trsv<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << uplo << "/" << t << "/" << diag << "/" << n;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void rocblas_trsv_f(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CHECK_ROCBLAS_STATUS(rocblas_strsv(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CHECK_ROCBLAS_STATUS(rocblas_dtrsv(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, rocblas_handle& rb_handle, std::string uplo,
+         std::string t, std::string diag, index_t n, bool* success) {
+  // Standard test setup.
+  const char* uplo_str = uplo.c_str();
+  const char* t_str = t.c_str();
+  const char* diag_str = diag.c_str();
+
+  index_t xlen = n;
+  index_t lda = n;
+  index_t incX = 1;
+
+  // The counters are double. We convert n to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  double n_d = static_cast<double>(n);
+
+  state.counters["n"] = n_d;
+
+  // Compute the number of A non-zero elements.
+  const double A_validVal = .5 * n_d * (n_d + 1);
+
+  double nflops_tot = 2.0 * A_validVal;
+  state.counters["n_fl_ops"] = nflops_tot;
+
+  {
+    double mem_readA = A_validVal;
+    double mem_readX = n_d;
+    double mem_writeX = n_d;
+    state.counters["bytes_processed"] =
+        (mem_readA + mem_readX + mem_writeX) * sizeof(scalar_t);
+  }
+
+  // Matrix options (rocBLAS)
+  const rocblas_fill uplo_rb =
+      uplo_str[0] == 'u' ? rocblas_fill_upper : rocblas_fill_lower;
+  const rocblas_diagonal diag_rb =
+      diag_str[0] == 'u' ? rocblas_diagonal_unit : rocblas_diagonal_non_unit;
+  const rocblas_operation trans_rb =
+      t_str[0] == 'n' ? rocblas_operation_none : rocblas_operation_transpose;
+
+  // Data sizes
+  const int m_size = lda * n;
+  const int v_size = 1 + (xlen - 1) * incX;
+
+  // Input matrix/vector, output vector.
+  std::vector<scalar_t> m_a(m_size);
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(v_size);
+
+  // Populate the main diagonal with larger values.
+  for (index_t i = 0; i < n; ++i)
+    for (index_t j = 0; j < n; ++j)
+      m_a[(i * lda) + i] = (i == j) ? blas_benchmark::utils::random_scalar(
```

Comment:
`m_a[( i * lda) + i]` seems to override diagonal value once they are set. Can you check that this is what we want? 
I think that something like `m_a[(i * lda) + j]`  would update only the main diagonal.

Reply:
Of course thanks for spotting that!